Migrate figwheel conf from project.clj to figwheel ns

voxlet opened the pull request:
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

Fixes #2999

### Summary:

[comment]: # (Summarise the problem and how the pull request solves it)
Replaced the figwheel profile :cljsbuild section with the equivalent in figwheel.edn format (along with the config reshaping code in figwheel-api), making sure not to change the behavior of `figwheel-api/start`.

### Review notes (optional):

I erred on the conservative side with the changes, preserving the *-test builds and warning-handler in the android build.

### Steps to test:
- Confirm `lein figwheel-repl` behavior has not changed

status: ready

voxlet committed:
Migrate figwheel conf from project.clj to figwheel ns

Signed-off-by: Julien Eluard <julien.eluard@gmail.com>

ghost commented:
Hey @voxlet, thanks for making your first pull request in status-react! :heart:

jeluard commented on env/dev/figwheel.clj:
```diff
@@ -0,0 +1,47 @@
+
+(defn get-build [builds id]
+  (when-let [build (first (filter #(= (:id %) (get-id id)) builds))]
+    (if (test-id? id)
```
Is this part still relevant? If not we could simply remove it.

voxlet replied:
Not that I can see, at least in terms of immediate code dependency. It was kept for the case of everyday REPL session use, but as far as I can tell the *-test runs are already broken in develop. 

Maybe we can go ahead and delete the code, and see if there any complaints? I think the chances of that happening are low enough.

jeluard replied:
Please remove it then! :)

jeluard reviewed (commented):
Thanks looks great!

dmitryn reviewed (approved):
Thanks for the contribution!

voxlet commented:
@jeluard Thanks for the feedback, removed the test builds. 
Also migrated the key names to the external figwheel.edn format (described [here](https://github.com/bhauman/lein-figwheel/blob/0f62d6d043abb6156393fd167f6c1496c5439689/sidecar/src/figwheel_sidecar/schemas/config.clj#L47-L74)), and moved the static nREPL port config there as well.

flexsurfer commented:
@voxlet please squash commits
